### PR TITLE
fix: remove calico installation step in Makefile

### PR DESCRIPTION
**Pull request by chewong**
Signed-off-by: Ernest Wong <chuwon@microsoft.com>

 

 
**What type of PR is this?**

**What this PR does / why we need it**:

Follow-up PR for https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1287. Looks like we are now installing calico twice on the workload cluster, causing the following flake in https://testgrid.k8s.io/provider-azure-master-signal#capz-azure-disk:

```log
Warning: resource serviceaccounts/calico-kube-controllers is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
serviceaccount/calico-kube-controllers configured
Error from server (AlreadyExists): error when creating "templates/addons/calico.yaml": deployments.apps "calico-kube-controllers" already exists
```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [x] squashed commits
- [x] includes documentation
- [x] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -496,8 +496,6 @@ create-workload-cluster: $(ENVSUBST)
 		elif [[ "${CLUSTER_TEMPLATE}" == *windows* ]]; then \
 			echo "windows being applied" \
 			kubectl --kubeconfig=./kubeconfig apply -f templates/addons/windows; \
-		else \
```
Comment (CecileRobertMichon) [inline Makefile:499]: what about windows and ipv6 test templates? are those using cluster resource sets too?

Comment (chewong) [reply Makefile:499]: Looks like they both are. Looks like I can completely remove the if statement?

Comment (CecileRobertMichon) [reply Makefile:499]: If they're no longer needed let's remove the entire if statement.

I wonder if the conformance tests have the CRS as well though, running one below to confirm

**Comment by CecileRobertMichon** [comment]
@jsturtevant this windows deployment timed out after 900.115s 😢 

/retest
/approve

**Pull request merged**